src/proxy/http/remap/NextHopConsistentHash.cc

```diff
@@ -57,17 +57,23 @@ std::shared_ptr<HostRecord>
 NextHopConsistentHash::chashLookup(const std::shared_ptr<ATSConsistentHash> &ring, uint32_t cur_ring, ParentResult &result,
                                    HttpRequestData &request_info, bool *wrapped, uint64_t sm_id)
 {
-  uint64_t               hash_key = 0;
-  ATSHash64Sip24         hash;
+  uint64_t hash_key = 0;
+  // Create hash instance based on configured algorithm and seeds
+  std::unique_ptr<ATSHash64> hash;
+  if (hash_algorithm == "siphash13") {
```

Comment:
Looks like it's almost a duplicate of `ParentConsistentHash::createHashInstance`. Can we have just one place to do this?